[DOC-599][TA] Add Tech advisory TA-25193 (#25507)
yugabyte/yugabyte-db: docs/content/preview/releases/techadvisories/ta-25193.md

The main cause is the premature removal of retention barriers on the database. This issue affects both gRPC and logical replication models.

## Mitigation

The existing CDC stream/replication slot will become unusable and needs to be deleted/dropped. A new CDC stream/replication slot should be created, and the snapshot should be consumed before starting to stream real-time changes to avoid any data loss. 

To create a new CDC stream/replication slot, choose from one of the following options:

- If the application does not require the previous values of a row, use before image/replica identity CHANGE.
- If the application requires the previous values of a row, switch to before image mode ALL or replica identity FULL.

## Details

By default, YugabyteDB has a history retention of 15 minutes (configured by `timestamp_history_retention_interval_sec`) on the database. Furthermore, if the assigned before image mode or replica identity mandates further retention, then a hybrid time-based retention barrier is set for CDC. This hybrid time barrier is moved periodically based on the client's acknowledgment.

> The issue due to the logic that checked if the hybrid time-based retention barrier could be moved was only based on whether the before image mode was set to ALL or the replica identity was set to FULL. For all other before-image modes and replica identities that required further retention, these barriers were lifted prematurely.

Reviewer: ```suggestion
The issue was caused by the logic that checked whether the hybrid time-based retention barrier could be moved. This logic only checked whether the before image mode was set to ALL or the replica identity was set to FULL. For all other before-image modes and replica identities that required further retention, these barriers were lifted prematurely.
```
